service/src/main/java/bio/terra/cbas/dao/MethodDao.java

```diff
@@ -112,4 +114,26 @@ public int countMethods(String methodName, String methodVersion) {
 
     return jdbcTemplate.queryForObject(sql, params, Integer.class);
   }
+
+  public int createGithubMethodSourceDetails(GithubMethodDetails details) {
+    return jdbcTemplate.update(
+        "insert into github_method_details (repository, organization, path, private, method_id) "
+            + "values (:repository, :organization, :path, :_private, :methodId)",
+        new BeanPropertySqlParameterSource(details));
+  }
+
+  public GithubMethodDetails getMethodSourceDetails(UUID methodId) {
+    String sql =
+        "SELECT * FROM github_method_details WHERE github_method_details.method_id = :methodId";
+
+    MapSqlParameterSource params = new MapSqlParameterSource(Map.of("methodId", methodId));
+
+    return jdbcTemplate.queryForObject(sql, params, new GithubMethodSourceDetailsMapper());
+  }
+
+  public int deleteMethodSourceDetails(UUID methodId) {
```

Comment:
I was going back an forth about that myself... I think maybe yes? Since we may have many new sources in the future, it may be nice to have a separate Dao associated with each new source.